core/src/vaults.ts

```diff
@@ -381,6 +390,51 @@ const getLatestVault = async (network: string, walletAddress: string) => {
     return new BigNumber(lastHex._hex).toNumber();
 };
 
+export const openVaultWithProxiedContractAndDrawDebt = async (
+    network: string,
+    proxyAddress: string,
+    collateralType: CollateralType,
+    collateralAmount: BigNumber,
+    debtAmountDai: BigNumber,
+    proxyType = 'CROPPER'
+) => {
+    const config = getCollateralConfigByType(collateralType);
+    giveAllowanceToAddress(network, config.symbol, proxyAddress, collateralAmount);
+
+    const signer = await getSigner(network);
+    const proxyContract = new ethers.Contract(proxyAddress, DS_PROXY, signer);
+    const method = getMethodSignature('openLockGemAndDraw(address,address,address,bytes32,uint256,uint256)');
+    const jugContractAddress = await getContractAddressByName(network, 'MCD_JUG');
+    const joinContractCollateralAddress = await getContractAddressByName(
+        network,
+        getJoinNameByCollateralType(collateralType)
+    );
+    const joinContractDaiAddress = await getContractAddressByName(network, 'MCD_JOIN_DAI');
+    const args = [
+        jugContractAddress,
+        joinContractCollateralAddress,
+        joinContractDaiAddress,
+        ethers.utils.formatBytes32String(collateralType),
+        collateralAmount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(),
+        debtAmountDai.shiftedBy(DAI_NUMBER_OF_DIGITS).toFixed(),
+    ];
+    const typesArray = ['address', 'address', 'address', 'bytes32', 'uint256', 'uint256'];
+    const encodedArgs = ethers.utils.defaultAbiCoder.encode(typesArray, args);
+    const transactionData = method + encodedArgs.substring(2);
+    const target = (await getContract(network, `PROXY_ACTIONS_${proxyType}`)).address;
+    const transaction = await proxyContract['execute(address,bytes)'](target, transactionData);
+    const events = await extractEventFromTransaction(
+        network,
+        transaction.hash,
+        'NewCdpRegistered(address,address,uint256)'
+    );
+    if (events.length !== 1) {
+        throw new Error('Unexpected number of NewCdpRegistered events');
+    }
+    const vaultId = new BigNumber(events[0].topics[3]).toNumber();
```

Comment:
This is again a fragile code. I am not sure that order of the events or topics is guaranteed by the blockchain and/or by the ethers, please follow the suggestion here https://github.com/sidestream-tech/unified-auctions-ui/pull/510#discussion_r1009469950